runtime/caml/fail.h

```diff
@@ -81,41 +81,46 @@ CAMLextern value caml_raise_if_exception(value res);
 extern "C" {
 #endif
 
-CAMLnoret CAMLextern void caml_raise (value bucket);
-
+/* The following functions raise immediately into OCaml. */
+CAMLnoret CAMLextern void caml_raise (value exception);
 CAMLnoret CAMLextern void caml_raise_constant (value tag);
-
 CAMLnoret CAMLextern void caml_raise_with_arg (value tag, value arg);
-
-CAMLnoret CAMLextern
-void caml_raise_with_args (value tag, int nargs, value arg[]);
-
+CAMLnoret CAMLextern void caml_raise_with_args (value tag,
+                                                int nargs, value arg[]);
 CAMLnoret CAMLextern void caml_raise_with_string (value tag, char const * msg);
-
 CAMLnoret CAMLextern void caml_failwith (char const *msg);
-
 CAMLnoret CAMLextern void caml_failwith_value (value msg);
-
 CAMLnoret CAMLextern void caml_invalid_argument (char const *msg);
-
 CAMLnoret CAMLextern void caml_invalid_argument_value (value msg);
-
 CAMLnoret CAMLextern void caml_raise_out_of_memory (void);
-
 CAMLnoret CAMLextern void caml_raise_stack_overflow (void);
-
 CAMLnoret CAMLextern void caml_raise_sys_error (value);
-
 CAMLnoret CAMLextern void caml_raise_end_of_file (void);
-
 CAMLnoret CAMLextern void caml_raise_zero_divide (void);
-
 CAMLnoret CAMLextern void caml_raise_not_found (void);
-
 CAMLnoret CAMLextern void caml_array_bound_error (void);
-
 CAMLnoret CAMLextern void caml_raise_sys_blocked_io (void);
 
+/* Non-raising variants of the above functions. The exception is
+   returned as a normal value, which can be raised with [caml_raise],
+   typically after cleaning-up resources. */
```

Comment:
I think it would make sense to also suggest `Result_exception` to return a `caml_result` instead of raising.